spec/extensions/fourier_transform_functions.md

```diff
@@ -0,0 +1,506 @@
+-   Optional parameters must be [keyword-only](https://www.python.org/dev/peps/pep-3102/) arguments.
+-   Broadcasting semantics must follow the semantics defined in {ref}`broadcasting`.
+-   Unless stated otherwise, functions must support the data types defined in {ref}`data-types`.
+-   Unless stated otherwise, functions must adhere to the type promotion rules defined in {ref}`type-promotion`.
+-   Unless stated otherwise, floating-point operations must adhere to IEEE 754-2019.
+
+## Objects in API
+
+<!-- NOTE: please keep the functions in alphabetical order -->
+(function-fft)=
+### fft(a, /, *, n=None, axis=-1, norm='backward')
+
+Computes the one-dimensional discrete Fourier transform.
+
+#### Parameters
+
+-   **a**: _&lt;array&gt;_
+
+    -   Input array.
+
+-   **n**: _int_
+
+    -   Length of the transformed axis of the output. If it is `None`, the length of the input along the axis given by the `axis` keyword. Default: `None`.
+
+-   **axis**: _int_
+
+    -  Axis used to compute the Fourier transform. If it is not specified, the last axis is used. Default: `-1`.
+
+-   **norm**: _str_
+
+    -   Specify the normalization mode. Should be one of the following modes:
+
+        - `'backward'`: No normalization.
+        - `'ortho'`: Normalize by `1/sqrt(n)`
+        - `'forward'`: Normalize by `1/n`.
+
+        Default: `'backward'`
+
+#### Returns
+
+-   **out**: _&lt;array&gt;_
+
+    -   An array transformed along the axis indicated by the `axis` keyword.
+
+#### Raises
+
+-   If `axis` is larger than the last axis of `a`.
+
+(function-fftfreq)=
+### fftfreq(n, /, *, d=1.0)
+
+Returns the discrete Fourier transform sample frequencies. For a Fourier transform of length `n` and length unit of `d` the frequencies are described as:
+
+```
+f = [0, 1, ..., n/2-1, -n/2, ..., -1] / (d*n) if n is even
+f = [0, 1, ..., (n-1)/2, -(n-1)/2, ..., -1] / (d*n) if n is odd
+```
+
+#### Parameters
+
+-   **n**: _int_
+
+    -   Window length.
+
+-   **d**: _float_
+
+    -   Sample spacing between individual samples of the Fourier transform input. Default: `1.0`.
+
+#### Returns
+
+-   **out**: _&lt;array&gt;_
+
+    -   An array of length `n` containing the sample frequencies.
+
+(function-fftn)=
+### fftn(a, /, *, s=None, axes=None, norm='backward')
+
+Computes the n-dimensional discrete Fourier transform.
+
+#### Parameters
+
+-   **a**: _&lt;array&gt;_
+
+    -   Input array.
+
+-   **s**: _Union\[ Sequence\[ int ], Tuple\[ int, ... ] ]_
+
+    -   Size of each transformed axis of the output. If given, each axis will be either zero-padded or trimmed to the length `s[i]` before computing the fast Fourier transform. Otherwise, no padding will be performed in each dimension. Default: `None`.
```

Comment:
```suggestion
    -   Size of each transformed axis of the output. If given, each axis will be either zero-padded or trimmed to the length `s[i]` before computing the Fourier transform. Otherwise, the length of the input along the axis given by the `axes` keyword. Default: `None`.
```